shesmu-pluginapi/src/main/java/ca/on/oicr/gsi/shesmu/plugin/filter/AlertFilter.java

```diff
@@ -300,16 +324,41 @@ private static <S> Parser parse3(
 
   private boolean negate;
 
-  public abstract <F> F convert(AlertFilterBuilder<F, String> f);
+  /**
+   * Converts this filter to an implementation-specific form
+   *
+   * @param builder the filter builder to use
+   * @return the filter implementation
+   * @param <F> the result filter type
+   */
+  public abstract <F> F convert(AlertFilterBuilder<F, String> builder);
 
+  /**
+   * Checks if the filter is inverted
+   *
+   * @return true if the filter should be inverted
+   */
   public boolean isNegate() {
     return negate;
   }
 
+  /**
+   * Convience method to construct a filter based on negation status
```

Comment:
Convenience

Reply:
Oh, let me know before you merge so I can squash this and any other changes.

Reply:
squash pls!